### PR TITLE
Feature sb 15 sign in with google

### PR DESCRIPTION
**Pull request by VladManukian**
1. Added google login endpoint in auth-api
2. Separated the logic of Google sign in and sign up
3. Fix sonarjs rule in eslint config

### **General**
- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [x] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file
- [x] Used camelCase for variables and functions
- [x] Date and time formats are on the constants
- [x] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [ ] covered by tests
- [x] Check your commit messages meet
### Frontend
- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the
- [x] No text in the components, use i18n approach
- [x] No inline styles
- [x] Imports are absolute
- [x] Attach a screenshot if PR has visual changes.

### CODE REVIEW

File: src/pages/auth/components/sign-in-form.tsx [hunk 1]
```diff
@@ -1,19 +1,80 @@
...
 
 import { useSignInForm } from "../hooks/index.ts";
 import { AuthLinks, ForgotPasswordLink, SignUpLink } from "./index.ts";
 import { StyledFormContainer } from "./styles.ts";
 
 const SignInForm: React.FC = () => {
+	const dispatch = useAppDispatch();
+	const navigate = useNavigate();
 	const { t } = useTranslation();
+	const {
```
Comment (lyudad) [inline src/pages/auth/components/sign-in-form.tsx:24]: empty space above

File: src/pages/auth/components/sign-in-form.tsx [hunk 2]
```diff
@@ -1,19 +1,80 @@
...
+		control,
+		errors,
+		handleFormSubmit,
+		isLoading,
+		serverError,
+		setServerError,
+	} = useSignInForm();
+	const [
```
Comment (lyudad) [inline src/pages/auth/components/sign-in-form.tsx:33]: empty space above

File: src/pages/auth/components/sign-up-form.tsx [hunk 3]
```diff
@@ -1,23 +1,74 @@
...
-	onClick: (credentialResponse: CredentialResponse) => void;
-};
-
-const SignUpForm: React.FC<Properties> = ({ onClick }: Properties) => {
+const SignUpForm: React.FC = () => {
+	const dispatch = useAppDispatch();
+	const navigate = useNavigate();
 	const { t } = useTranslation();
 	const { control, errors, handleFormSubmit, isLoading, serverError } =
 		useSignUpForm();
```
Comment (lyudad) [inline src/pages/auth/components/sign-up-form.tsx:24]: why its on new line?

File: src/pages/auth/components/sign-up-form.tsx [hunk 4]
```diff
@@ -1,23 +1,74 @@
...
+	const [
```
Comment (lyudad) [inline src/pages/auth/components/sign-up-form.tsx:27]: empty line above

File: src/pages/auth/components/sign-up-form.tsx (hunk 3 above)
Comment (VladManukian) [reply src/pages/auth/components/sign-up-form.tsx:24]: Because of the rules eslint[prettier/prettier]
Should we delete it?

Comment (VladManukian) [reply src/pages/auth/components/sign-up-form.tsx:24]: We have 	"printWidth": 80,
If we change it now to 100 or 120, we should format all the code. I leave this question open.

**Pull request merged**